### PR TITLE
Remove warnings: cast between incompatible function types

### PR DESCRIPTION
**Pull request by rbuj**
Bump glib-2 required version to 2.58:

-  G_SOURCE_FUNC
  https://gitlab.gnome.org/GNOME/glib/commit/039fa6897bb4c5cd30caec8ee3add1a4ef612f76

```
gs-manager.c:390:43: warning: cast between incompatible function types from ‘void (*)(void *)’ to ‘void (*)(void *, void *)’ [-Wcast-function-type]
  390 |   g_slist_foreach (manager->priv->themes, (GFunc)g_free, NULL);
      |                                           ^
--
gs-prefs.c:205:35: warning: cast between incompatible function types from ‘void (*)(void *)’ to ‘void (*)(void *, void *)’ [-Wcast-function-type]
  205 |   g_slist_foreach (prefs->themes, (GFunc)g_free, NULL);
      |                                   ^
--
gs-prefs.c:603:35: warning: cast between incompatible function types from ‘void (*)(void *)’ to ‘void (*)(void *, void *)’ [-Wcast-function-type]
  603 |   g_slist_foreach (prefs->themes, (GFunc)g_free, NULL);
      |                                   ^
--
mate-screensaver-preferences.c:810:25: warning: cast between incompatible function types from ‘void (*)(void *)’ to ‘void (*)(void *, void *)’ [-Wcast-function-type]
  810 |  g_list_foreach (files, (GFunc) (g_object_unref), NULL);
      |                         ^
--
copy-theme-dialog.c:437:41: warning: cast between incompatible function types from ‘void (*)(void *)’ to ‘void (*)(void *, void *)’ [-Wcast-function-type]
  437 |  g_slist_foreach (dlg->priv->all_files, (GFunc) (g_object_unref), NULL);
      |                                         ^
copy-theme-dialog.c:439:45: warning: cast between incompatible function types from ‘void (*)(void *)’ to ‘void (*)(void *, void *)’ [-Wcast-function-type]
  439 |  g_slist_foreach (dlg->priv->all_basenames, (GFunc) (g_free), NULL);
      |                                             ^
--
mate-screensaver-dialog.c:348:14: warning: cast between incompatible function types from ‘gboolean (*)(void)’ {aka ‘int (*)(void)’} to ‘gboolean (*)(void *)’ {aka ‘int (*)(void *)’} [-Wcast-function-type]
  348 |   g_idle_add((GSourceFunc) quit_response_ok, NULL);
      |              ^
mate-screensaver-dialog.c:366:15: warning: cast between incompatible function types from ‘gboolean (*)(void)’ {aka ‘int (*)(void)’} to ‘gboolean (*)(void *)’ {aka ‘int (*)(void *)’} [-Wcast-function-type]
  366 |    g_idle_add((GSourceFunc) response_request_quit, NULL);
      |               ^
--
mate-screensaver-dialog.c:598:13: warning: cast between incompatible function types from ‘gboolean (*)(void)’ {aka ‘int (*)(void)’} to ‘gboolean (*)(void *)’ {aka ‘int (*)(void *)’} [-Wcast-function-type]
  598 |  g_idle_add((GSourceFunc) popup_dialog_idle, NULL);
      |             ^
--
gs-lock-plug.c:2144:62: warning: cast between incompatible function types from ‘void (*)(GSLockPlug *)’ to ‘gboolean (*)(void *)’ {aka ‘int (*)(void *)’} [-Wcast-function-type]
 2144 |  plug->priv->datetime_timeout_id = g_timeout_add_seconds (1, (GSourceFunc) date_time_update, plug);
      |                                                              ^
--
gs-auth-pam.c:667:37: warning: cast between incompatible function types from ‘int (*)(int)’ to ‘void * (*)(void *)’ [-Wcast-function-type]
  667 |                                     (GThreadFunc) gs_auth_thread_func,
      |                                     ^
```

### CODE REVIEW

File: src/gs-lock-plug.c [hunk 1]
```diff
@@ -2141,7 +2141,7 @@ gs_lock_plug_init (GSLockPlug *plug)
 		date_time_update (plug);
 		gtk_widget_show_all (plug->priv->vbox);
 	}
-	plug->priv->datetime_timeout_id = g_timeout_add_seconds (1, (GSourceFunc) date_time_update, plug);
+	plug->priv->datetime_timeout_id = g_timeout_add_seconds (1, G_SOURCE_FUNC (date_time_update), plug);
```
Comment (monsta) [inline src/gs-lock-plug.c]: I'm not sure it's correct. According to docs, the callback should be a function returning `gboolean` to determine if it should be called again or not.

File: src/mate-screensaver-dialog.c [hunk 2]
```diff
@@ -345,7 +345,7 @@ static gboolean auth_check_idle(GSLockPlug* plug)
 	if (res)
 	{
 		again = FALSE;
-		g_idle_add((GSourceFunc) quit_response_ok, NULL);
+		g_idle_add (G_SOURCE_FUNC (quit_response_ok), NULL);
```
Comment (monsta) [inline src/mate-screensaver-dialog.c]: This function takes no arguments, but here it's casted to a function accepting one `gpointer`. Same for two more code changes below...

Comment (rbuj) [reply src/mate-screensaver-dialog.c]: https://developer.gnome.org/glib/stable/glib-The-Main-Event-Loop.html#G-SOURCE-FUNC:CAPS

Comment (rbuj) [reply src/mate-screensaver-dialog.c]: Do you mean something like this?
```diff
diff --git a/src/mate-screensaver-dialog.c b/src/mate-screensaver-dialog.c
index 2074a13..57335b3 100644
--- a/src/mate-screensaver-dialog.c
+++ b/src/mate-screensaver-dialog.c
@@ -104,7 +104,7 @@ static void response_ok(void)
        fflush(stdout);
 }
 
-static gboolean quit_response_ok(void)
+static gboolean quit_response_ok(gpointer user_data)
 {
        response_ok();
        gtk_main_quit();
@@ -345,7 +345,7 @@ static gboolean auth_check_idle(GSLockPlug* plug)
        if (res)
        {
                again = FALSE;
-               g_idle_add (G_SOURCE_FUNC (quit_response_ok), NULL);
+               g_idle_add (quit_response_ok, NULL); /* g_idle_add (GSourceFunc quit_response_ok, gpointer user_data) */
        }
        else
        {
```

Comment (rbuj) [reply src/mate-screensaver-dialog.c]: I suppose you mean https://github.com/mate-desktop/mate-screensaver/pull/222

File: src/gs-lock-plug.c (hunk 1 above)
Comment (rbuj) [reply src/gs-lock-plug.c]: I suppose you mean https://github.com/mate-desktop/mate-screensaver/pull/223

**Pull request merged**